xhyve/xhyve.go

```diff
@@ -603,22 +603,21 @@ func (d *Driver) extractKernelImages() error {
 	}
 
 	volumeRootDir := d.ResolveStorePath(isoMountPath)
-	vmlinuz := filepath.Join(volumeRootDir, "boot", d.Vmlinuz)
-	initrd := filepath.Join(volumeRootDir, "boot", d.Initrd)
 
-	log.Debugf("Extracting %s into %s", d.Vmlinuz, d.ResolveStorePath(d.Vmlinuz))
-	if err := mcnutils.CopyFile(vmlinuz, d.ResolveStorePath(d.Vmlinuz)); err != nil {
-		return err
-	}
-
-	log.Debugf("Extracting %s into %s", d.Initrd, d.ResolveStorePath(d.Initrd))
-	if err := mcnutils.CopyFile(initrd, d.ResolveStorePath(d.Initrd)); err != nil {
-		return err
-	}
+	defer func() {
+		log.Debugf("Unmounting %s", isoFilename)
+		err = hdiutil("detach", volumeRootDir)
+	}()
 
-	log.Debugf("Unmounting %s", isoFilename)
-	if err := hdiutil("detach", volumeRootDir); err != nil {
-		return err
+	for _, f := range []string{"vmlinuz64", "bzImage", "initrd", "initrd.img"} {
```

Comment:
Ah, sorry confused.
minikube and boot2docker are used `bzImage` and `vmlinuz64`.
But generally,
- kenrnel filename: `vmlinuz` or `vmlinux` or `bzImage`
- initrd filename: `initrd` or `initrd.gz`

e.g.
http://ftp.debian.org/debian/dists/jessie/main/installer-amd64/current/images/hd-media/

So, I think 
```go
[]string{"vmlinux", "vmlinuz64", "vmlinuz", "bzImage", "initrd", "initrd.gz", "initrd.img"}
```
is better for feature use.

Do you get what I want to say...?